src/ints/mouse.cpp

```diff
@@ -88,6 +88,7 @@ static struct {
 	Bit16u hidden;
 	float add_x,add_y;
 	Bit16s min_x,max_x,min_y,max_y;
+	Bit16s max_screen_x,max_screen_y;
```

Comment:
The game sets max x/y to the screen dimensions minus the cursor size. I assume it's so that the game can just avoid having to clip the cursor against the bottom and right sides of the screen. But it does draw the cursor on the screen coordinate system, so for the host to guest cursor integration to work correctly, the separate variables are there for the INT 33h emulation to round up the max dimensions to the actual screen instead.

For example in Daggerfall, the cursor will follow the host correctly if:

(given by Daggerfall through INT 33h)
max_x = 312 max_y = 192
(computed by DOSBox-X INT 33h handling)
max_screen_x = 319 max_screen_y = 199

If max_x (312) and max_y (192) are used for host to guest mapping, the cursor will move slightly off from the host on the bottom right hand corner of the screen.